### ISSUE TITLE
[HorizontalListView] On Carousel mode, keep animation when scroll is disabled

### ISSUE THREAD

**Issue by chaparo** [roubachof/Sharpnado.Presentation.Forms#35]
Hello,

We would have a new feature on the carousel mode to animate scrolling when the DisableScroll property is set, which means that this property only disables gestures.

We have already developed this, do you think that could be a good feature for your HorizontalListView (carousel mode)?

Thank you

**Comment by roubachof** [comment]
I don't really understand...
Do you mean scrolling is disabled and you have a swipe gesture to navigate to the next or previous view (like regular basic carousel) ?

**Comment by chaparo** [comment]
The scrolling is disabled, but we would like to keep the animation when we scroll programatically (we change CurrentIndex property)

**Comment by roubachof** [comment]
ok I see. Like a ```AnimateCurrentIndexChange``` property ?

**Comment by chaparo** [comment]
Yes, we called it KeepAnimation but I think you get the idea.

**Comment by chaparo** [comment]
```csharp        
private void ScrollToCurrentItem()
{
    if (Element.CurrentIndex == -1 || Control.GetAdapter() == null || Element.CurrentIndex >= Control.GetAdapter().ItemCount)
    {
        return;
    }

    int offset = 0;
    if (HorizontalLinearLayoutManager != null)
    {
        int itemWidth = PlatformHelper.Instance.DpToPixels(Element.ItemWidth + Element.ItemSpacing);
        int width = Control.MeasuredWidth;

        switch (Element.SnapStyle)
        {
            case SnapStyle.Center:
                offset = (width / 2) - (itemWidth / 2);
                break;
        }

        if (Element.DisableScroll && Element.KeepAnimation)
        {
            HorizontalLinearLayoutManager.CanScroll = true;
            RecyclerView.SmoothScroller smoothScroller = new CustomLinearSmoothScroller(this.Context);
            smoothScroller.TargetPosition = Element.CurrentIndex;
            HorizontalLinearLayoutManager.StartSmoothScroll(smoothScroller);
            return;
        }
    }

    LinearLayoutManager?.ScrollToPositionWithOffset(Element.CurrentIndex, offset);
}
```

**Comment by roubachof** [comment]
Also: why allow it only in DisableScroll mode ? Could work in all situations

**Comment by chaparo** [comment]
You're right, that mean it allow to dissociate the animation from the gesture ?

**Comment by chaparo** [comment]
But if I remember we made this, because we need to allow the CanScroll in Android, otherwise it's not working.

in the IosRender we just change this part and it true that we could only use the KeepAnimation property
```csharp
            Control.ScrollToItem(
                NSIndexPath.FromRowSection(Element.CurrentIndex, 0),
                position,
                !Element.DisableScroll || Element.KeepAnimation);
```

but in Android it's a bit more tricky.

**Comment by roubachof** [comment]
mmmh interesting, would be quite easy to integrate...
I will be releasing a new update for the xamarin forms sharpnado lib including some "pure" xamarin forms tabs (bottom bar, underlined tab, custom tabs), will see if I can squeeze this in :)

**Comment by chaparo** [comment]
that's would be perfect ! 
In any case your component is really awesome !

**Comment by roubachof** [comment]
@chaparo sorry man, I had too much on my plate for this release...
But I would be happy to take a look at a PR :)

**Comment by chaparo** [comment]
Ok I'll submit that this week.

**Comment by chaparo** [comment]
Hello,
We are checking your new release. 
The ViewSwitcher matches more with our feature (so we might not use anymore the HorizotalListview with this keepAnimation feature ;) ).
First, we use the HorizontalListView, because it implements the recycler view.
We notice, the ViewSwitcher use the LazyView, which if I understand correctly, on the loading page the tabs are not loaded. But we couldn't find were the Dispose where called on the SwitcherView.

For example we have a form with 13 tabs containing different views. 
We are a little worried about performances between 13 items in an ViewSwitcher vs HorizontalListView.

What do you think about that ?

**Comment by roubachof** [comment]
Hey!

The ViewSwitcher doesn't recycle views, so if you have 13 views which are all the same, I would still recommend to use the HLV.

The ViewSwitcher supports LazyView, which is another concerpt I will blog about in the next weeks, but it's available now, and there is a working example in the silly app (you just have to uncomment xaml in the SillyBottomTabsPage.xaml.
The LazyView are only loaded when you access them the first time, so it will speed up loading time and memory vs a regular view. But if you load them all, it will still contain 13 different views.

So I guess if you want recycling, your best match is HLV with Carousel mode linked to a TabHostView

**Comment by roubachof** [comment]
Errr no you cannot use TabHostView with 13 tabs, sorry for that :)

**Comment by chaparo** [comment]
Ok, thank you :) 

About the HLV, we don't have same template for all views, we use a DataTemplateSelector which gives us the good template from the CurrentIndex.

Do you think that this behavior could induce some issue in your HLV ?

**Comment by roubachof** [comment]
Nope no problem :) the HLV supports DataTemplateSelector, and recycles them according to the DataTemplate type.
We have an issue with that on iOS (it recycles all view the same way), but it will be quickly fixed since the PR is in process.
